nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -422,12 +415,95 @@ buildOperatorCodes(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
   return fbb.CreateVector(fb_op_codes);
 }
 
+/**
+ * @brief get Serialized Tensor shape
+ *
+ * @param dim Tensor dimension to convert
+ * @param type tflite Tensor type
+ * @param effective_dimension bitflag of valid 'nchw' dimension, ex) 0b1011
+ * means,nhw is effective
+ * @param is_signature check if the given dimension is regarded as a tensor
+ * signature
+ * @return flatbuffers::Offset<flatbuffers::Vector<int32_t>>
+ */
+flatbuffers::Offset<flatbuffers::Vector<int32_t>>
+buildTesorShape(const TensorDim &dim, flatbuffers::FlatBufferBuilder &fbb,
+                int effective_dimension, bool is_signature = false) {
+  std::vector<int32_t> fb_dimension;
+
+  /// batch dimension is always considered with signature
+  if ((effective_dimension >> 3) & 1) {
+    fb_dimension.push_back(is_signature ? -1 : dim.batch());
+  }
+
+  for (unsigned int i = 1; i < MAXDIM; ++i) {
+    if (((effective_dimension >> (MAXDIM - i - 1)) & 1) == 0) {
+      continue;
+    }
+    fb_dimension.push_back(dim.getTensorDim(i));
+  }
+
+  return fbb.CreateVector(fb_dimension);
+}
+
+flatbuffers::Offset<flatbuffers::Vector<flatbuffers::Offset<tflite::Tensor>>>
+buildTensors(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
+  /// @todo: the actual (suqeezed) tensor dimension must be known before coming
+  /// here. For now, it is directly guessed for the fc layer
+  const auto &variables = map.getIndexMap<const Var_Grad *>().getData();
+  const auto &buffer_map = map.getIndexMap<const float *, TfOpIdxMap::Buffer>();
+
+  std::vector<flatbuffers::Offset<tflite::Tensor>> fb_tensors;
+  fb_tensors.reserve(variables.size());
+
+  auto create_tensor = [&fbb, &buffer_map](const Var_Grad *var) {
+    /// @fixme: There is a fixed assumption that effective_dimension is 0b1001,
+    /// this is not true, this should be supported inherently from TensorDim
+    /// class.
+    int FIX_ME_FLAG = 0b1001;
+    auto shape = buildTesorShape(var->getDim(), fbb, FIX_ME_FLAG, false);
+    auto shape_sig = buildTesorShape(var->getDim(), fbb, FIX_ME_FLAG, true);
```

Comment:
can you add some comment as how shape is different from shape_sig here?

Reply:
Updated